### ISSUE TITLE
anomaly: lof score fluctuation

### ISSUE THREAD

**Issue by TkrUdagawa** [jubatus/jubatus#1221]
I was testing the performance of add API and found that the lof was fluctuating for the identical data set.

My test was below:

Start `jubaanomaly` server with a config of `lsh`, hash_num 64.

Next, I made a datum list containing 3000 datums. Each datum has 100 random numbers.

Then repeat the process below 5 times:
1. clear Jubatus server
2. add all datums
3. calc lof score of the last datum in the list

The scores were 
```
1st time: 1.0249063968658447
2nd time: 1.0471123456954956
3rd time: 1.0471124649047852
4th time: 1.0249063968658447
5th time: 1.0471123456954956
```
The detail codes and results are [here](https://gist.github.com/TkrUdagawa/acb6fa40234a6b368bcf644199ae788b).

We need to detect the cause of this fluctuation.

**Comment by TkrUdagawa** [comment]
Now I understand what was happening.
This problem was caused by the sort order of the points which have the identical similarity score to the query.

Changing id by 100, I printed out the neighbor points and their similarity scores used for the lof calculation(neighbor_num=5).

```
1st (id:100 ~ id: 3099)
nearest neighbor points are:
id: 3099 0
id: 661 0.125
id: 575 0.125
id: 165 0.125
id: 2363 0.140625

2nd(id:200 ~ id: 3199)
nearest neighbor points are:
id: 3199 0
id: 761 0.125
id: 675 0.125
id: 265 0.125
id: 324 0.140625
```
The 5th nearest id differs between 1st and 2nd.
The id corresponding to id:324 is 224 in the 1st.
But 224 comes after 2363 in the 1st because the ids are sorted by dictionary order.

These differences may cause the fluctuation of lof scores.
